src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs

```diff
@@ -332,9 +357,23 @@ async Task<T> IDurableOrchestrationContext.CreateTimer<T>(DateTime fireAt, T sta
                 expirationTime: fireAt,
                 isReplay: this.InnerContext.IsReplaying);
 
+            this.IsLongRunningTimer = false;
+
             return result;
         }
 
+        public string ValidOutOfProcTimer(DateTime fireAt)
```

Comment:
General approach for these types of problems is to do something like 

`bool ValidateOutOfProcTimer(DateTIme fireAt, out string errorMessage)`.

That way the caller doesn't have to know that an empty string means that it is valid, and a non-empty string means it isn't.